ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.cpp

```diff
@@ -0,0 +1,111 @@
+// [WriteFile Name=ConfigureBasemapStyleLanguage, Category=Maps]
+// [Legal]
+// Copyright 2024 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifdef PCH_BUILD
+#include "pch.hpp"
+#endif // PCH_BUILD
+
+#include "Basemap.h"
+#include "BasemapStyleParameters.h"
+#include "ConfigureBasemapStyleLanguage.h"
+#include "Error.h"
+#include "Map.h"
+#include "MapQuickView.h"
+#include "MapTypes.h"
+#include "Point.h"
+#include "SpatialReference.h"
+#include "Viewpoint.h"
+
+#include <QFuture>
+
+using namespace Esri::ArcGISRuntime;
+
+ConfigureBasemapStyleLanguage::ConfigureBasemapStyleLanguage(QObject* parent /* = nullptr */) :
+  QObject(parent)
+{
+  basemapStyleParameters = new BasemapStyleParameters(this);
+}
+
+ConfigureBasemapStyleLanguage::~ConfigureBasemapStyleLanguage() = default;
+
+void ConfigureBasemapStyleLanguage::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<ConfigureBasemapStyleLanguage>("Esri.Samples", 1, 0, "ConfigureBasemapStyleLanguageSample");
+}
+
+MapQuickView* ConfigureBasemapStyleLanguage::mapView() const
+{
+  return m_mapView;
+}
+
+// Set the view (created in QML)
+void ConfigureBasemapStyleLanguage::setMapView(MapQuickView* mapView)
+{
+  m_map = new Map(this);
```

Comment:
Seems like we should initialize the spatial reference of the map, since we are immediately setting it on a map view and then setting a viewpoint. I'm actually surprised it works to set a viewpoint when there's no spatial reference set.

```suggestion
  m_map = new Map(SpatialReference::webMercator());
```

Reply:
Also, I would move the Map initialization to the constructor